opentelemetry-sdk/src/logs/mod.rs

```diff
@@ -57,4 +58,51 @@ mod tests {
             .expect("Attributes are expected");
         assert_eq!(attributes.len(), 2);
     }
+
+    #[test]
+    fn logging_sdk_shutdown_test() {
+        // Arrange
+        let exporter: InMemoryLogsExporter = InMemoryLogsExporter::default();
+        let logger_provider = LoggerProvider::builder()
+            .with_log_processor(SimpleLogProcessor::new(Box::new(exporter.clone())))
+            .build();
+        set_logger_provider(logger_provider);
+
+        // Act
+        let logger = logger("test-logger");
+        let mut log_record: LogRecord = LogRecord::default();
+        log_record.severity_number = Some(Severity::Error);
+        log_record.severity_text = Some("Error".into());
+        let attributes = vec![
+            (Key::new("key1"), "value1".into()),
+            (Key::new("key2"), "value2".into()),
+        ];
+        log_record.attributes = Some(attributes);
+        logger.emit(log_record);
+
+        // Intentionally *not* calling shutdown/flush
+        // on the provider, but instead relying on
+        // shutdown_logger_provider which causes
+        // the global provider to be dropped, and
+        // the sdk logger provider's drop implementation
+        // will cause shutdown to be called on processors/exporters.
+        shutdown_logger_provider();
+
+        // Assert
+        let exported_logs = exporter
+            .get_emitted_logs()
+            .expect("Logs are expected to be exported.");
+        assert_eq!(exported_logs.len(), 1);
+        let log = exported_logs
+            .first()
+            .expect("Atleast one log is expected to be present.");
+        assert_eq!(log.instrumentation.name, "test-logger");
+        assert_eq!(log.record.severity_number, Some(Severity::Error));
+        let attributes: Vec<(Key, AnyValue)> = log
+            .record
+            .attributes
+            .clone()
+            .expect("Attributes are expected");
+        assert_eq!(attributes.len(), 2);
```

Comment:
Should we validate for exported logs before doing shutdown, as SimpleLogProcessor will export them synchronously. And to ensure that shutdown has really happened, try calling emit() on existing logger instance. There shouldn't be any exporter happening this time.

Reply:
thats a good point. I should use batchprocessor here instead. Or a custom one, which never publishes logs to exporter, except when shutdown/flush called - i think this is the best way to test